kubernetes/resource_kubernetes_affinity_test.go

```diff
@@ -183,31 +188,32 @@ func TestAccKubernetesPod_with_pod_anti_affinity_with_preferred_during_schedulin
-					resource.TestCheckResourceAttr("kubernetes_pod.test", "spec.0.affinity.0.pod_anti_affinity.0.preferred_during_scheduling_ignored_during_execution.0.pod_affinity_term.0.label_selector.0.match_expressions.0.values.#", "2"),
-					resource.TestCheckResourceAttr("kubernetes_pod.test", "spec.0.affinity.0.pod_anti_affinity.0.preferred_during_scheduling_ignored_during_execution.0.pod_affinity_term.0.label_selector.0.match_expressions.0.values.1996459178", "bar"),
-					resource.TestCheckResourceAttr("kubernetes_pod.test", "spec.0.affinity.0.pod_anti_affinity.0.preferred_during_scheduling_ignored_during_execution.0.pod_affinity_term.0.label_selector.0.match_expressions.0.values.2356372769", "foo"),
-					resource.TestCheckResourceAttr("kubernetes_pod.test", "spec.0.affinity.0.pod_anti_affinity.0.preferred_during_scheduling_ignored_during_execution.0.pod_affinity_term.0.label_selector.0.match_labels.%", "0"),
-					resource.TestCheckResourceAttr("kubernetes_pod.test", "spec.0.affinity.0.pod_anti_affinity.0.preferred_during_scheduling_ignored_during_execution.0.pod_affinity_term.0.namespaces.#", "0"),
-					resource.TestCheckResourceAttr("kubernetes_pod.test", "spec.0.affinity.0.pod_anti_affinity.0.preferred_during_scheduling_ignored_during_execution.0.pod_affinity_term.0.topology_key", "kubernetes.io/hostname"),
-					resource.TestCheckResourceAttr("kubernetes_pod.test", "spec.0.affinity.0.pod_anti_affinity.0.preferred_during_scheduling_ignored_during_execution.0.weight", "100"),
+					resource.TestCheckResourceAttr("kubernetes_pod.test", fmt.Sprintf("%s.#", keyName), "1"),
+					resource.TestCheckResourceAttr("kubernetes_pod.test", fmt.Sprintf("%s.0.pod_affinity_term.#", keyName), "1"),
+					resource.TestCheckResourceAttr("kubernetes_pod.test", fmt.Sprintf("%s.0.pod_affinity_term.0.label_selector.#", keyName), "1"),
+					resource.TestCheckResourceAttr("kubernetes_pod.test", fmt.Sprintf("%s.0.pod_affinity_term.0.label_selector.0.match_expressions.#", keyName), "1"),
+					resource.TestCheckResourceAttr("kubernetes_pod.test", fmt.Sprintf("%s.0.pod_affinity_term.0.label_selector.0.match_expressions.0.key", keyName), "security"),
+					resource.TestCheckResourceAttr("kubernetes_pod.test", fmt.Sprintf("%s.0.pod_affinity_term.0.label_selector.0.match_expressions.0.operator", keyName), "NotIn"),
+					resource.TestCheckResourceAttr("kubernetes_pod.test", fmt.Sprintf("%s.0.pod_affinity_term.0.label_selector.0.match_expressions.0.values.#", keyName), "2"),
+					resource.TestCheckResourceAttr("kubernetes_pod.test", fmt.Sprintf("%s.0.pod_affinity_term.0.label_selector.0.match_expressions.0.values.0", keyName), "bar"),
+					resource.TestCheckResourceAttr("kubernetes_pod.test", fmt.Sprintf("%s.0.pod_affinity_term.0.label_selector.0.match_expressions.0.values.0", keyName), "foo"),
+					resource.TestCheckResourceAttr("kubernetes_pod.test", fmt.Sprintf("%s.0.pod_affinity_term.0.label_selector.0.match_labels.%%", keyName), "0"),
```

Comment:
This is an escape character for `fmt.Sprintf`: `%`. I had to use it so I could literally print out `%` here like the other tests do. So that's why there are two of them.